### PR TITLE
Add connection tree favorites

### PR DESCRIPTION
**Pull request by farosch**
Implemented a favorites menu in the connection tree window.
Replaces #1305 

## Description
- Added a new drop down menu in the conenction tree window
- Introduced new local connection property
- Introduced new inherited connection property

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: mRemoteNGTests/Connection/AbstractConnectionInfoDataTests.cs [hunk 1]
```diff
@@ -405,6 +405,15 @@ public void MacAddressNotifiesOnValueChange()
 
         [Test]
         public void UserFieldNotifiesOnValueChange()
+        {
+            var wasCalled = false;
+            _testAbstractConnectionInfoData.PropertyChanged += (sender, args) => wasCalled = true;
+            _testAbstractConnectionInfoData.Favorite = true;
```
Comment (sparerd) [inline mRemoteNGTests/Connection/AbstractConnectionInfoDataTests.cs]: I think you transposed these two tests when copying. The `UserFieldNotifiesOnValueChange` test is setting the `Favorite` setting, and the `FavoriteNotifiesOnValueChange` test is setting the `UserField` setting

File: mRemoteV1/Config/Serializers/ConnectionSerializers/MsSql/LocalConnectionPropertiesModel.cs [hunk 2]
```diff
@@ -16,5 +16,10 @@ public class LocalConnectionPropertiesModel
         /// Indicates whether this container is expanded in the tree
         /// </summary>
         public bool Expanded { get; set; }
+
+        /// <summary>
+        /// Indicates whether this container is expanded in the tree
```
Comment (sparerd) [inline mRemoteV1/Config/Serializers/ConnectionSerializers/MsSql/LocalConnectionPropertiesModel.cs]: Update comment

**Comment by sparerd** [comment]
@farosch All looks good to me.

What would you think about having a `Favorites` folder within the quick connect dropdown too?
image

**Comment by farosch** [comment]
@sparerd That's a good idea, haven't had thought about that.
I've added the favorites to the quick connect dropdown:
image

**Review by sparerd — APPROVED** [review]

**Comment by sparerd** [comment]
Cool, that looks good! Merging

**Comment by junkhacker** [comment]
With the new favorites feature, I'd love to have a favorites toolbar. the quickconnect menu with favorites is nice, but I'd love to put buttons for a single click to connect on the toolbar area so i don't have to use the drop down menu for the handful of connections i use the most.

**Comment by Kvarkas** [comment]
its planed to add work belt feature in next release. you can drag drop (or mark) connections what you want to upper in work belt and access is by duble click will be availbale

**Pull request merged**